#CLASS A AIS POSITION REPORT (MESSAGES 1, 2, AND 3)
#A Class A AIS unit broadcasts the following information every 2 to 10 seconds while underway, and every 3 minutes while at anchor at a power level of 12.5 watts.

Header header

int32 id   #Identifier for this message 1, 2 or 3

int32 repeat_indicator   #Used by the repeater to indicate how many times a message has been repeated. See Section 4.6.1, Annex 2; 0-3; 0 = default; 3 = do not repeat any more.

int32 mmsi   #MMSI number


#0 = under way using engine, 1 = at anchor, 2 = not under command, 3 = restricted maneuverability, 4 = constrained by her draught, 5 = moored, 6 = aground, 7 = engaged in fishing, 8 = under way sailing, 9 = reserved for future amendment of navigational status for ships carrying DG, HS, or MP, or IMO hazard or pollutant category C, high speed craft (HSC), 10 = reserved for future amendment of navigational status for ships carrying dangerous goods (DG), harmful substances (HS) or marine pollutants (MP), or IMO hazard or pollutant category A, wing in ground (WIG); 11 = power-driven vessel towing astern (regional use); 12 = power-driven vessel pushing ahead or towing alongside (regional use);
# 13 = reserved for future use,
# 14 = AIS-SART (active), MOB-AIS, EPIRB-AIS
# 15 = undefined = default (also used by AIS-SART, MOB-AIS and EPIRB-AIS under test)
int32 nav_status

#0 to +126 = turning right at up to 708 deg per min or higher
# 0 to -126 = turning left at up to 708 deg per min or higher Values between 0 and 708 deg per min coded by ROTAIS = 4.733 SQRT(ROTsensor) degrees per min
# where ROTsensor is the Rate of Turn as input by an external Rate of Turn Indicator (TI). ROTAIS is rounded to the nearest integer value.
# +127 = turning right at more than 5 deg per 30 s (No TI available)
# -127 = turning left at more than 5 deg per 30 s (No TI available)
# -128 (80 hex) indicates no turn information available (default).
# ROT data should not be derived from COG information.
string rot_over_range
float64 rot

#Speed over ground in 1/10 knot steps (0-102.2 knots)
# 1 023 = not available, 1 022 = 102.2 knots or higher
float64 sog

#The position accuracy (PA) flag should be determined in accordance with the table below:
# 1 = high (<= 10 m)
# 0 = low (> 10 m)
# 0 = default
int32 position_accuracy


#Latitude in 1/10 000 min (+/-90 deg, North = positive (as per 2's complement), South = negative (as per 2's complement). 91deg (3412140h) = not available = default)
float64 latitude

#Longitude in 1/10 000 min (+/-180 deg, East = positive (as per 2's complement), West = negative (as per 2's complement).
# 181= (6791AC0h) = not available = default)
float64 longitude

#Course over ground in 1/10 = (0-3599). 3600 (E10h) = not available = default. 3 601-4 095 should not be used
float64 cog

#Degrees (0-359) (511 indicates not available = default)
int32 true_heading

#UTC second when the report was generated by the electronic position system (EPFS) (0-59, or 60 if time stamp is not available,
#which should also be the default value, or 61 if positioning system is in manual input mode, or 62 if electronic position fixing
#system operates in estimated (dead reckoning) mode, or 63 if the positioning system is inoperative)
int32 timestamp

#0 = not available = default
# 1 = not engaged in special maneuver
# 2 = engaged in special maneuver
# (i.e.: regional passing arrangement on Inland Waterway)
int32 special_manoeuvre

#Not used. Should be set to zero. Reserved for future use.
int32 spare

string raim
int32 sync_state
int32 slot_timeout
float64 utc_hour
float64 utc_min
float64 utc_spare